src/app/guest-book/components/new-message-dialog/new-message-dialog.component.ts

```diff
@@ -0,0 +1,83 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { FormControl, ReactiveFormsModule, Validators } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatDialogModule, MatDialogRef } from '@angular/material/dialog';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { Store } from '@ngrx/store';
+import { GuestBookMessage } from '../../models/guestBookMessage';
+import { GuestBookActions } from '../../state/guest-book.actions';
+import { GuestBookMessageState } from '../../state/guest-book.reducer';
+
+@Component({
+  selector: 'app-new-contact-dialog',
+  templateUrl: './new-message-dialog.component.html',
+  styleUrls: ['./new-message-dialog.component.scss'],
+  standalone: true,
+  imports: [
+    MatDialogModule,
+    MatFormFieldModule,
+    MatInputModule,
+    MatButtonModule,
+    MatIconModule,
+    ReactiveFormsModule,
+    CommonModule
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NewMessageDialogComponent {
+  errorMessage$ = this.store.select(GuestBookActions.createGuestBookMessageFailure);
+  name = new FormControl('', [Validators.required]);
```

Comment:
Usually it's a good idea to group controls in a FormGroup.
That allows to simplify bulk operations - e.g. in the template:
[disabled]="name.invalid || email.invalid || message.invalid"
could be replaced with 
[disabled]="formGroup.invalid"